src/main/java/org/jenkinsci/plugins/workflow/steps/BodyExecutionCallback.java

```diff
@@ -112,19 +112,20 @@ public void onFailure(StepContext context, Throwable t) {
         @Override public final void onSuccess(StepContext context, Object result) {
             try {
                 finished(context);
-                context.onSuccess(result);
```

Comment:
No need to catch exceptions from `onSuccess` itself; these should not happen, but would be thrown up the call chain.